editor/plugins/org.fusesource.ide.camel.editor/src/org/fusesource/ide/camel/editor/preferences/UserLabelsListEditor.java

```diff
@@ -0,0 +1,134 @@
+/******************************************************************************* 
+ * Copyright (c) 2016 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.fusesource.ide.camel.editor.preferences;
+
+import static org.fusesource.ide.camel.model.service.core.model.AbstractCamelModelElement.USER_LABEL_DELIMETER;
+import static org.fusesource.ide.camel.model.service.core.model.AbstractCamelModelElement.USER_LABEL_REGEX;
+
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.dialogs.IInputValidator;
+import org.eclipse.jface.dialogs.InputDialog;
+import org.eclipse.jface.preference.ListEditor;
+import org.eclipse.jface.window.Window;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Widget;
+import org.fusesource.ide.camel.editor.internal.UIMessages;
+
+/**
+ * 
+ * @author Andrej Podhradsky (apodhrad@redhat.com)
+ *
+ */
+public class UserLabelsListEditor extends ListEditor {
+
+	public static final IInputValidator VALIDATOR = new IInputValidator() {
+		@Override
+		public String isValid(String newText) {
+			if (newText.matches(USER_LABEL_REGEX)) {
+				return null;
+			}
+			return UIMessages.userLabels_errorMessage;
+		}
+	};
+
+	private Button editButton;
+
+	public UserLabelsListEditor(String name, String labelText, Composite parent) {
+		super(name, labelText, parent);
+	}
+
+	@Override
+	protected String[] parseString(String string) {
+		if (string == null || string.equals("")) {
```

Comment:
that's only syntax sugar ;-)

Regarding the functionality, it could be used only with non-uri components. For any endpoint we display its uri which is the only one reasonable information to be displayed.

So, we need to improve the information about what it actually does. And yeah... the error message could be better.

Regarding the ID, I'd like to remove this option. I don't think it is useful to display the IDs.

Thank you for your fast feedback. I'll do some changes and let you know once I'm done